sdk/python/tests/test_materialize_from_bigquery_to_datastore.py

```diff
@@ -78,16 +74,72 @@ def test_bigquery_ingestion_correctness(self):
 
         # run materialize()
         fs.materialize(
-            ["test_bq_correctness"],
+            [fv.name],
             datetime.utcnow() - timedelta(minutes=5),
             datetime.utcnow() - timedelta(minutes=0),
         )
 
         # check result of materialize()
-        entity_key = EntityKeyProto(
-            entity_names=["driver_id"], entity_values=[ValueProto(int64_val=1)]
+        response_dict = fs.get_online_features(
+            [f"{fv.name}:value"], [{"driver_id": 1}]
+        ).to_dict()
+        assert abs(response_dict[f"{fv.name}:value"][0] - checked_value) < 1e-6
+
+    def test_bigquery_query_to_datastore_correctness(self):
+        # create dataset
+        ts = pd.Timestamp.now(tz="UTC").round("ms")
+        checked_value = (
+            random.random()
+        )  # random value so test doesn't still work if no values written to online store
+        data = {
+            "id": [1, 2, 1],
+            "value": [0.1, 0.2, checked_value],
+            "ts_1": [ts - timedelta(minutes=2), ts, ts],
+            "created_ts": [ts, ts, ts],
+        }
+        df = pd.DataFrame.from_dict(data)
+
+        # load dataset into BigQuery
+        job_config = bigquery.LoadJobConfig()
+        table_id = f"{self.gcp_project}.{self.bigquery_dataset}.query_correctness_{int(time.time())}"
+        query = f"SELECT * FROM `{table_id}`"
+        job = self.client.load_table_from_dataframe(df, table_id, job_config=job_config)
+        job.result()
+
+        # create FeatureView
+        fv = FeatureView(
+            name="test_bq_query_correctness",
+            entities=["driver_id"],
+            features=[Feature("value", ValueType.FLOAT)],
```

Comment:
By the way, neither your tests (nor my historical retrieval tests) provide coverage of all types that we support. We should add explicit tests for all types as cards into our backlog.